### PR TITLE
Support running all docker images locally

### PR DESCRIPTION
**Pull request by dennisschagt**
Resolves https://github.com/newsboat/newsboat/issues/2474

Changes based on [a stackoverflow post](https://stackoverflow.com/questions/71918710/allow-docker-container-host-user-to-write-on-bind-mounted-host-directory).

### CODE REVIEW

File: docker/code-formatting-tools.dockerfile [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #   docker build \
 #       --tag=newsboat-code-formatting-tools \
 #       --file=docker/code-formatting-tools.dockerfile \
-#       docker
+#       .
```
Comment (Minoru) [inline docker/code-formatting-tools.dockerfile:12]: Why is this needed? It doesn't look like Dockerfiles use anything outside the docker directory.

Comment (dennisschagt) [reply docker/code-formatting-tools.dockerfile:12]: This is not strictly necessary, however, I prefer to keep these consistent between all of our docker files.
In the buildtools docker files, I've used a `COPY` command to pull in the new `entrypoint.sh` so there it is was necessary to update the path (cirrus seems to always specify the root directory of the git checkout, without option to override it so I did have to include the `docker/` part in the `COPY` command)

File: docker/ubuntu_18.04-build-tools.dockerfile [hunk 2]
```diff
@@ -37,8 +37,10 @@
 #
 #   docker run \
 #       --rm \
-#       --mount type=bind,source=$(pwd),target=/home/builder/src \
-#       --user $(id -u):$(id -g) \
+#       --mount type=bind,source=$(pwd),target=/src \
+#       --user root
```
[the remark this answers is not in the document]
Comment (dennisschagt) [reply docker/ubuntu_18.04-build-tools.dockerfile:41]: The approach taken in this PR requires the entrypoint script to update the user/group id of the `builder` user.
We need to be root to do that.
However, in the end, we want to run the build commands as the `builder` user.

The main issue is that cirrus does not expose a way to specify a user, so it will use the user as specified by the last `USER` statement in the docker file.
Cirrus also does not run the command specified by `ENTRYPOINT`.
That makes it necessary to specify the `builder` user at the end of the docker file.
However, locally, we do want to run the entrypoint script as root to fixup the user/group id in the container to match the user/group id of the host user.
That is why the `--user root` option is necessary at runtime.

**Comment by dennisschagt** [comment]
Closing this in favor of https://github.com/newsboat/newsboat/pull/2729
Still, thanks for the review!
